session.go

```diff
@@ -43,6 +44,10 @@ type Session interface {
 	// used it will return nil.
 	PublicKey() PublicKey
 
+	Context() context.Context
```

Comment:
Sigh. The problem is the Session interface is already sooooo huuuuuge and it's only going to get bigger with signals and I also added Permissions. Interfaces should be small. A lot of the rest of what's in ssh.Context does qualify as "to inform, not to control" and is more for maintainers than users (at least how I interpret it). So it might feel inconsistent, but at the moment I'm ok with the tradeoff.